src/hearings/pipes/show-hide.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {Observable} from 'rxjs';
+import {AbstractHiddenConverter} from '../converters/abstract.hidden.converter';
+import {DefaultHiddenConverter} from '../converters/default.hidden.converter';
+import {WelshHiddenConverter} from '../converters/welsh.hidden.converter';
+import {IsHiddenSource} from '../models/hearings.enum';
+import * as fromHearingStore from '../store';
+
+@Pipe({
+  name: 'isHidden'
+})
+export class ShowHidePipe implements PipeTransform {
+
+  constructor(protected readonly hearingStore: Store<fromHearingStore.State>) {
```

Comment:
Do we need to inject the store into a pipe?. Pipe are meant for simple data transformations and pure.

Reply:
I think that's fine we are using pipe to subscribe data in the store, pipe can also do complex things it's not exactly for simple data transformation.

Reply:
Yes pipes can do complicated things but in general should be simple and pure. Given the same input they will should return the same result. We should at least pass the state of the store into the `transform()` method not inject the whole store.

Reply:
I can't get your point or I don't agree that pass the state into the transform() method, it will be then very heavy on the html and its specific ts. It's not a problem to inject the store into the pipe, coz that's how that store is injected or how it is used, it could be unsubscribed it from the pipe on OnDestroy if any subscription is made, but giving that we are using aync pipe it has no side effect. Also if you inject the store that is just an reference to the store, it's not actually load all the data on the pipe.

Reply:
This is complicated for no reason. Your pipe responsibility is to hide or show, but your pipe has side effects and contains state. Pipes are transformational functions. In your case the pipe transform should take hearing state and return true or false.  E.g. html like this 
`<div *ngIf="section.isHiddenSource | isHidden:  hearingState"></div>`

ts 
`public transform(isHiddenSource: IsHiddenSource, state: HearingState): boolean {
    let converter: AbstractHiddenConverter = new DefaultHiddenConverter(state);
    switch (isHiddenSource) {
      case IsHiddenSource.WELSH_LOCATION:
        converter = new WelshHiddenConverter(state);
        break;
      default:
        break;
    }
    return converter.transformHidden();
  }`

Reply:
This is not completed at all, store from the pipe here is an immutable single source of data, I can't get that why you said it has side effect, we are not dispatching data in the pipe, we are just using async pipe to show the data.
In you sample, it pass the state from the html that makes no benefit at all, it increases the complexity in the html and its type script. and compare the lines of code it seems more complex.
This approach has been reviewed and approved by Andy.
Unless you could specify what the 'side effect' is we will stick with this solution.

Reply:
here is an example for inject store in pipe:
https://stackoverflow.com/questions/59130911/is-it-recommended-to-subscribe-to-the-store-ngrx-inside-a-pipe